libraries/AP_OSD/AP_OSD_MAX7456.h

```diff
@@ -39,7 +39,21 @@ class AP_OSD_MAX7456 : public AP_OSD_Backend
     void clear() override;
 
     // return a correction factor used to display angles correctly
-   float get_aspect_ratio_correction() const override;
+    float get_aspect_ratio_correction() const override;
+
+    bool is_compatible_with_backend_type(AP_OSD::osd_types type) const override {
+        switch(type) {
+        case AP_OSD::osd_types::OSD_MAX7456:
+        case AP_OSD::osd_types::OSD_SITL:
+            return false;
+        default:
```

Comment:
Don't use a `default:` here; enumerate the other types.  That forces someone creating a new backend type to actually specify the compatability here

Similarly elsewhere.

Reply:
ok, I had to add a "return false" at the end for the compiler would complain about unreachable code but this way if the enum is expanded the compiler will complain as well so good to go.